### DISCUSSION TITLE
Fuck open source garbage is so tiring

### DISCUSSION THREAD

**Discussion by vovanhau123** [citizenfx/fivem#2842] [Engineering Discussion]
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[2/2] ⡀ ffi-napi
error C:\Windows\System32\fivem\ext\native-doc-tooling\node_modules\ffi-napi: Command failed.
Exit code: 1
Command: node-gyp-build
Arguments:
Directory: C:\Windows\System32\fivem\ext\native-doc-tooling\node_modules\ffi-napi
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.1
gyp info using node@12.22.12 | win32 | x64
gyp info find Python using Python version 3.12.1 found at "C:\Program Files\Python312\python.exe"
gyp info find VS using VS2022 (17.12.35323.107) found at:
gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Preview"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Program Files\Python312\python.exe
gyp info spawn args [
gyp info spawn args   'C:\\Users\\levan\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Windows\\System32\\fivem\\ext\\native-doc-tooling\\node_modules\\ffi-napi\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\levan\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\levan\\AppData\\Local\\node-gyp\\Cache\\12.22.12\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\levan\\AppData\\Local\\node-gyp\\Cache\\12.22.12',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\levan\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\levan\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\12.22.12\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\Windows\\System32\\fivem\\ext\\native-doc-tooling\\node_modules\\ffi-napi',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\Windows\\System32\\fivem\\ext\\native-doc-tooling\\node_modules\\ffi-napi\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\MSBuild.exe
gyp info spawn args [
gyp info spawn args   'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64'
gyp info spawn args ]
MSBUILD : error MSB1009: Project file does not exist.
Switch: build/binding.sln
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\levan\AppData\Local\Yarn\Data\global\node_modules\node-gyp\lib\build.js:203:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Windows_NT 10.0.22631
gyp ERR! command "C:\\Windows\\System32\\fivem\\ext\\native-doc-gen\\node.exe" "C:\\Users\\levan\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Windows\System32\fivem\ext\native-doc-tooling\node_modules\ffi-napi
gyp ERR! node -v v12.22.12
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok

**Comment by magi200** [comment]
remove yarn in LocalAppdata and prebuild

**Reply by kerneljay** [reply]
Any help? No module named "distutils" while prebuild

**Reply by kerneljay** [reply]
https://gist.github.com/kerneljay/67be85c90041576a780088df5d08445e
